src/uwtools/tests/drivers/test_ungrib.py

```diff
@@ -0,0 +1,184 @@
+    assert not dst.is_symlink()
+    driverobj.gribfile_aaa()
+    assert dst.is_symlink()
+
+
+def test_Ungrib_namelist_wps(driverobj):
+    src = driverobj._rundir / "namelist.wps.in"
+    with open(src, "w", encoding="utf-8") as f:
+        yaml.dump({}, f)
+    dst = driverobj._rundir / "namelist.wps"
+    assert not dst.is_file()
+    driverobj._driver_config["namelist_file"] = {"base_file": src}
+    driverobj.namelist_wps()
+    assert dst.is_file()
+
+
+def test_Ungrib_provisioned_run_directory(driverobj):
+    with patch.multiple(
+        driverobj,
+        gribfile_aaa=D,
+        namelist_wps=D,
+        runscript=D,
+        vtable=D,
+    ) as mocks:
+        driverobj.provisioned_run_directory()
+    for m in mocks:
+        mocks[m].assert_called_once_with()
+
+
+def test_Ungrib_run_batch(driverobj):
+    with patch.object(driverobj, "_run_via_batch_submission") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_Ungrib_run_local(driverobj):
+    driverobj._batch = False
+    with patch.object(driverobj, "_run_via_local_execution") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_Ungrib_runscript(driverobj):
+    dst = driverobj._rundir / "runscript"
+    assert not dst.is_file()
+    driverobj._driver_config["execution"].update(
+        {
+            "batchargs": {"walltime": "01:10:00"},
+            "envcmds": ["cmd1", "cmd2"],
+            "mpicmd": "runit",
+            "threads": 8,
+        }
+    )
+    driverobj._config["platform"] = {"account": "me", "scheduler": "slurm"}
+    driverobj.runscript()
+    with open(dst, "r", encoding="utf-8") as f:
+        lines = f.read().split("\n")
+    # Check directives:
+    assert "#SBATCH --account=me" in lines
+    assert "#SBATCH --time=01:10:00" in lines
+    # Check environment commands:
+    assert "cmd1" in lines
+    assert "cmd2" in lines
+    # Check execution:
+    assert "test $? -eq 0 && touch %s/done" % driverobj._rundir
+
+
+def test_Ungrib_vtable(driverobj):
+    src = driverobj._rundir / "Vtable.GFS.in"
+    src.touch()
+    driverobj._driver_config["vtable"] = src
+    dst = driverobj._rundir / "Vtable.GFS"
+    assert not dst.is_symlink()
+    driverobj.vtable()
+    assert dst.is_symlink()
+
+
+def test_Ungrib__driver_config(driverobj):
+    assert driverobj._driver_config == driverobj._config["ungrib"]
+
+
+def test_Ungrib__resources(driverobj):
+    account = "me"
+    scheduler = "slurm"
+    walltime = "01:10:00"
+    driverobj._driver_config["execution"].update({"batchargs": {"walltime": walltime}})
+    driverobj._config["platform"] = {"account": account, "scheduler": scheduler}
+    assert driverobj._resources == {
+        "account": account,
+        "rundir": driverobj._rundir,
+        "scheduler": scheduler,
+        "walltime": walltime,
+    }
+
+
+def test_Ungrib__runscript_path(driverobj):
+    assert driverobj._runscript_path == driverobj._rundir / "runscript"
```

Comment:
```suggestion
    assert driverobj._runscript_path == driverobj._rundir / "runscript.ungrib"
```
if the driver code gets updated to follow the newer convention from `main` wherein runscript files have their component name as a suffix.